sdk/python/lib/test/test_next_serialize.py

```diff
@@ -84,41 +116,93 @@ async def test_dict(self):
         self.assertDictEqual({"a": 42, "b": 99}, prop)
 
     @pulumi_test
-    async def test_custom_resource(self):
-        fake_urn = "urn:pulumi:mystack::myproject::my:mod:Fake::fake"
-        res = MyCustomResource(fake_urn)
+    async def test_custom_resource_preview(self):
+        settings.SETTINGS.dry_run = True
+        rpc.register_resource_module("test", "index", MyResourceModule())
+        set_mocks(MyMocks())
+
+        res = MyCustomResource("test")
+        urn = await res.urn.future()
+        id = await res.id.future()
 
         settings.SETTINGS.feature_support["resourceReferences"] = False
         deps = []
         prop = await rpc.serialize_property(res, deps)
         self.assertListEqual([res], deps)
-        self.assertEqual("id", prop)
+        self.assertEqual(id, prop)
 
         settings.SETTINGS.feature_support["resourceReferences"] = True
         deps = []
         prop = await rpc.serialize_property(res, deps)
+        self.assertListEqual([res, res], deps)
```

Comment:
Why does the list contain two copies of `res`?

Reply:
b/c the marshaller marshals the URN and the ID of the resource. Both of these outputs depend on the resource, so each adds the resource to the list. We could (and should) improve this by using a set rather than a list until we need to send these over the wire.